managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/EntryCacheImpl.java

```diff
@@ -118,12 +118,14 @@ public boolean insert(EntryImpl entry) {
             entryBuf.readerIndex(readerIdx);
         }
 
-        if (entries.put(entry.getPosition(), new EntryImpl(entry.getPosition(), cachedData))) {
+        PositionImpl position = entry.getPosition();
+        EntryImpl cacheEntry = EntryImpl.create(position, cachedData);
```

Comment:
I forgot to release `cachedData` here. Since `cacheEntry` retains the buffer, we need to do `cachedData.release()` just after this line